module/zfs/bqueue.c

```diff
@@ -87,13 +87,14 @@ bqueue_enqueue(bqueue_t *q, void *data, uint64_t item_size)
 void *
 bqueue_dequeue(bqueue_t *q)
 {
-	void *ret;
+	void *ret = NULL;
 	uint64_t item_size;
 	mutex_enter(&q->bq_lock);
-	while (q->bq_size == 0) {
+	ret = list_remove_head(&q->bq_list);
+	while (ret == NULL && q->bq_size == 0) {
 		cv_wait(&q->bq_pop_cv, &q->bq_lock);
+		ret = list_remove_head(&q->bq_list);
 	}
-	ret = list_remove_head(&q->bq_list);
```

Comment:
This NULL deref could be triggered if the CV receives an unexpected wakeup and `bq_size` isn't an accurate count of list items.  That's impossible with the existing code since both are always updated under the mutex.  If it were to happen would mean there's a bug in the bqueue  implementation.  So for this case I'd suggest just adding an ASSERT here to make that invariant explicit.  Hopefully that will make Coverity happy.